### ISSUE TITLE
Swagger (v2) not deserialized properly

### ISSUE THREAD

**Issue by sbacquet** [Microsoft/OpenAPI.NET#390]
In the following code :

```
var stream = new HttpClient().GetStreamAsync("https://petstore.swagger.io/v2/swagger.json").Result;

var inputDocument = new OpenApiStreamReader().Read(stream, out var diagnostic);

Debug.Assert(diagnostic.SpecificationVersion == OpenApiSpecVersion.OpenApi2_0);

var schema = inputDocument.Paths["/pet/{petId}"].Operations[OperationType.Get].Responses["200"].Content["application/json"].Schema;
```
`schema` ends up null, whereas it should contain Pet.

**Comment by darrelmiller** [comment]
Hey @sbacquet   Good catch.

The problem comes because of the produces with two media types.  If you run the petstore OpenAPI through the workbench tool you get the following V3 output.

```yaml
        '200':
          description: successful operation
          content:
            application/xml:
              schema:
                $ref: '#/components/schemas/Pet'
            application/json: { }
```

The schema is preserved for the first content type but lost for the second.  I suspect this is a regression related to other changes around maintaining state whilst translating V2 into V3.

I will investigate.

**Comment by sbacquet** [comment]
Hello Darrel,

Thanks a lot for investigating.
May I suggest to add a unit test to cover this case ? ;)

Thanks again!

**Comment by huaxing-yuan** [comment]
Hello, i think i'm running into a similar scenario the response schema reference is empty.
The bug is introduced in the package v1.1.2. I rolled back to v1.1.1 the schema is now read correctly.

This is part of my swagger v2 document
```yaml
paths:
  /xxx:
    post:
      operationId: xxx
      produces:
        - application/json
      parameters:
        ***
      responses:
        '200':
          description: success
          schema:
            $ref: '#/definitions/checkOutput'
```

**Comment by darrelmiller** [comment]
@huaxing-yuan Thanks for the additional info, I suspected it was a regression.  And yes @sbacquet I'll make sure that our test cases are updated to account for this.

**Comment by VitaliyKurokhtin** [comment]
Should be resolved by #398

**Comment by darrelmiller** [comment]
This fix should be available in 1.1.3

**Comment by sbacquet** [comment]
I confirm it's fixed, thank you !
